fleetspeak/src/server/https/client_certificate_test.go

```diff
@@ -93,22 +119,27 @@ func makeTestClient(t *testing.T) (common.ClientID, *http.Client, []byte) {
 			ExpectContinueTimeout: 1 * time.Second,
 		},
 	}
-	return id, &cl, bc
+	return id, &cl, bc, clientCertFingerprint
 }
 
 func TestFrontendMode_MTLS(t *testing.T) {
 	ts := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		// test the valid frontend mode combination of receiving the client cert in the req
-		cert, err := GetClientCert(req, "", cpb.FrontendMode_MTLS)
+		cert, err := GetClientCert(req, "", cpb.FrontendMode_MTLS, "")
 		if err != nil {
 			t.Fatal(err)
 		}
 		// make sure we received the client cert in the req
 		if cert == nil {
 			t.Error("Expected client certificate but received none")
 		}
-		// test the invalid frontend mode combination
-		cert, err = GetClientCert(req, "", cpb.FrontendMode_HEADER_TLS)
+		// test the invalid frontend mode combination for frontend mode HEADER_TLS
+		cert, err = GetClientCert(req, "", cpb.FrontendMode_HEADER_TLS, "")
+		if err == nil {
+			t.Error("Expected error for invalid frontend mode combination but received none")
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures
We usually try to inform the maintainer, in a concise way, what we were testing, the input, what we got and then what we expected (or diffs, if it's long). So a common format (as the link shows) is like:
`FuncUnderTest(%v) = %v; want %v`

In this case, it's not really clear to me what's going on (we can discuss offline if you want)

Reply:
Thanks Tati, I converted them to table driven tests. Hope the added comments help describing the cases under test.